### PR TITLE
MRG: Export brainvision (raw), add .export method to mne.Evoked

### LINKED ISSUE #10678: Add BrainVision export via pybv
Just announcing that I plan to add export capabilities for BrainVision to the MNE-Python export suite. I thought we had discussed this in other issues already, but I can't find them or they have been closed.

I recently added a feature to pybv to write more complex events, so that mne.Annotations can now be properly written, see: https://github.com/bids-standard/pybv/pull/86

In the future I'd also like to be able to export electrode coordinates ("channel locations"), but I think that can come as an add on slightly later: https://github.com/bids-standard/pybv/issues/44

One feature apart from making this available to MNE-Python users would also be that we can refactor what we currently have in mne-bids: https://github.com/mne-tools/mne-bids/blob/fa7a94bb0ef03691aa881f39c90b23a7aa69a923/mne_bids/write.py#L963

Let me know if you have thoughts on this :-)

### PR DESCRIPTION
**Pull request by sappelhoff**
closes #10678 

This PR also:

- includes some cleanup of docstrs
- adds "export" as a method to mne.Evoked (currently was only there for raw and epochs)

to do:

- [x] fix pybv tests for annotations (some first_samp issue?)
- [x] fix evoked.export --> somehow not available for EvokedArray?

### CODE REVIEW

**Comment by sappelhoff** [comment]
@ephathaway I get MFF related "FutureWarnings" here that look like they'd need to be fixed upstream. Care to have a look? :-)

Try running the following on MNE-Python `main`:

- fails: `pytest mne/export/tests/test_export.py::test_export_evokeds_to_mff`
- passes: `pytest mne/export/tests/test_export.py::test_export_evokeds_to_mff -W ignore::FutureWarning`

**Comment by larsoner** [comment]
For now @sappelhoff in this PR you can add it as an ignore in `mne/conftest.py` or probably better as a `@pytest.filterwarnings` in the offending tests. Then a follow-up PR can remove them by fixing the issues properly

File: mne/export/tests/test_export.py [hunk 1]
```diff
@@ -396,6 +397,10 @@ def test_export_evokeds_to_mff(tmp_path, fmt, do_history):
     else:
         export_evokeds(export_fname, evoked, overwrite=True)
 
+    # test export from evoked directly
+    # XXX: this fails because "EvokedArray" has no export method?
+    # evoked[0].export(export_fname)
```
Comment (sappelhoff) [inline mne/export/tests/test_export.py:402]: @larsoner why does this not work for EvokedArray?

I added the `export` method to Evoked (see above) and wanted to add a quick test here ... but it fails :shrug: do I have to add the export method to some "super class"?

Comment (sappelhoff) [reply mne/export/tests/test_export.py:402]: unlike for raw and epochs, there is no "BaseEvoked"

Comment (larsoner) [reply mne/export/tests/test_export.py:437]: Evoked is effectively BaseEvoked, and EvokedArray inherits from it

https://github.com/mne-tools/mne-python/blob/a877fada44f4241e740dad3dee8ffeb1a50422e3/mne/evoked.py#L864

Knowing this, do you want to look further, or do you want me to look at the failure?

Comment (sappelhoff) [reply mne/export/tests/test_export.py]: Thanks! No need to look at the failure, this helps me.

**Pull request merged**